### PR TITLE
[Bug][Connector][FileBase]Parquet reader parsing array type exception.

### PR DESCRIPTION
**Pull request by lightzhao**
## Purpose of this pull request
Parquet reader parsing array type exception. [java.lang.ArrayStoreException]

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Review by TyrantLucifer — DISMISSED** [review]
Add ut/it cases to verify this pull request is valid

**Comment by lightzhao** [comment]
((GenericData.Array<?>) field).iterator().forEachRemaining(origArray::add)
image
The modified test result is：
image

**Comment by lightzhao** [comment]
@EricJoy2048 @TyrantLucifer @Hisoka-X PTAL, thanks.

**Comment by lightzhao** [comment]
who has time help review, thanks.

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/test/java/org/apache/seatunnel/connectors/seatunnel/file/writer/ParquetReadStrategyTest.java
Comment (EricJoy2048) [inline seatunnel-connectors-v2/connector-file/connector-file-base/src/test/java/org/apache/seatunnel/connectors/seatunnel/file/writer/ParquetReadStrategyTest.java]: Why not use assert?

Comment (lightzhao) [reply seatunnel-connectors-v2/connector-file/connector-file-base/src/test/java/org/apache/seatunnel/connectors/seatunnel/file/writer/ParquetReadStrategyTest.java]: done .PTAL.

**Comment by lightzhao** [comment]
please approve ci.

**Comment by lightzhao** [comment]
@Hisoka-X @EricJoy2048  @TyrantLucifer please approve ci.

**Comment by lightzhao** [comment]
@Hisoka-X @EricJoy2048 @TyrantLucifer please approve ci.

**Comment by ic4y** [comment]
Waiting for CI/CD.

**Comment by lightzhao** [comment]
@TaoZex @ic4y @Hisoka-X @EricJoy2048 @TyrantLucifer PTAL.

**Comment by lightzhao** [comment]
image
Why is there such an exception? It has nothing to do with the module I submitted, and the latest code has also been merged.

**Comment by lightzhao** [comment]
@EricJoy2048 @ic4y @TyrantLucifer PTAL.

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/test/resources/array.parquet
Comment (liugddx) [inline seatunnel-connectors-v2/connector-file/connector-file-base/src/test/resources/array.parquet:1]: Based on #5545, we need to remove the binary and automatically generate it

Comment (hailin0) [reply seatunnel-connectors-v2/connector-file/connector-file-base/src/test/resources/array.parquet:1]: +1

Comment (lightzhao) [reply seatunnel-connectors-v2/connector-file/connector-file-base/src/test/resources/array.parquet:1]: done.

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/test/java/org/apache/seatunnel/connectors/seatunnel/file/writer/ParquetReadStrategyTest.java
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-file/connector-file-base/src/test/java/org/apache/seatunnel/connectors/seatunnel/file/writer/ParquetReadStrategyTest.java]: Please use log instead of `System.out.println`.

Comment (lightzhao) [reply seatunnel-connectors-v2/connector-file/connector-file-base/src/test/java/org/apache/seatunnel/connectors/seatunnel/file/writer/ParquetReadStrategyTest.java]: ok

**Comment by hailin0** [comment]
LGTM

waiting for ci ok

**Comment by lightzhao** [comment]
> LGTM
> 
> waiting for ci ok

CI is a bit unstable, but it is successful after re-running.

**Comment by hailin0** [comment]
> > LGTM
> > waiting for ci ok
> 
> CI is a bit unstable, but it is successful after re-running.

Yes. Are you interested in helping improve it?

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/test/java/org/apache/seatunnel/connectors/seatunnel/file/writer/ParquetReadStrategyTest.java
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-file/connector-file-base/src/test/java/org/apache/seatunnel/connectors/seatunnel/file/writer/ParquetReadStrategyTest.java]: You can use `@DisabledOnOs(OS.WINDOWS)` to instead of this. Please refer https://github.com/apache/seatunnel/blob/6f49ec6ead82f842c6d93563ffce64853deb1bed/seatunnel-engine/seatunnel-engine-server/src/test/java/org/apache/seatunnel/engine/server/master/JobHistoryServiceTest.java#L39

**Comment by lightzhao** [comment]
> > > LGTM
> > > waiting for ci ok
> > 
> > 
> > CI is a bit unstable, but it is successful after re-running.
> 
> Yes. Are you interested in helping improve it?

ok, i  try to study it.

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/test/java/org/apache/seatunnel/connectors/seatunnel/file/writer/ParquetReadStrategyTest.java
Comment (lightzhao) [reply seatunnel-connectors-v2/connector-file/connector-file-base/src/test/java/org/apache/seatunnel/connectors/seatunnel/file/writer/ParquetReadStrategyTest.java]: ok

**Review by Hisoka-X — APPROVED** [review]
LGTM if test passed.

**Comment by lightzhao** [comment]
ci re-run success. @hailin0 @liugddx @ic4y PTAL.

**Comment by lightzhao** [comment]
@hailin0 @EricJoy2048 PTAL.

**Review by hailin0 — APPROVED** [review]

**Pull request merged**
